security-advisories/README.md

```diff
@@ -14,9 +14,178 @@ At this page we will list of all known security vulnerabilities found on OP-TEE.
+
+| Reported by  | CVE ID    | OP-TEE ID        | Affected versions  |
+| ------------ | :-------: | :--------------: | ------------------ |
+| [Riscure]    | Not/Ready | OP-TEE-2018-0009 | v3.3.0 and earlier |
+
+### Buffer checks missing when calling pseudo TAs
+The function `tee_svc_copy_param` is used to copy in parameters when a TA wants
+to open a session with or invoke a command upon another TA. It is used in
+system calls and is therefore indirectly callable by any TA. However, this
+function does not do sufficient parameter checking when the called TA is a
+pseudo TA. One of the parameters passed in is `callee_params` which is passed
+directly through from the TA. It is verified that this structure itself resides
+in either shared memory or memory which the calling TA has read access to.
+However, this structure can contain pointers as its members. The structure
+`callee_params` is first copied into the output parameter param. In the case
+that the called TA is a pseudo TA no further checking is done and a success
+code is returned. It is not verified that the members of param point to valid
+memory. This means there is a mismatch between the validation performed when
+invoking a normal TA and when invoking a pseudo TA. If a pseudo TA relies on
+the pointers being validated as it would be for a normal TA, it might use these
+pointers without further validation. This might result in memory corruption and
+disclosure.
+
+**optee_os.git:**
+ - [core: svc: always check ta parameters (d5c5b0b77b2)](
+https://github.com/OP-TEE/optee_os/commit/d5c5b0b77b2b589666024d219a8007b3f5b6faeb)
+
+| Reported by  | CVE ID    | OP-TEE ID        | Affected versions  |
+| ------------ | :-------: | :--------------: | ------------------ |
+| [Riscure]    | Not/Ready | OP-TEE-2018-0007 | v3.3.0 and earlier |
+
+### Potential disclosure of previously loaded TA code and data
+The function `elf_load_body` is used to load the code and data segments while
+dynamically loading a TA. The amount of memory allocated for the code and data
+segments is previously determined and the sum of it is stored in
+`state->vasize`. The actual allocated amount of memory is rounded up the next
+multiple of the memory pool granularity. To ensure that the newly loaded TA is
+not able to observe any data belonging to a TA previously stored on this exact
+location in memory, the memory block is set to zero. The size used to memset
+the block to zero is the sum of the sizes of the segments, not the rounded size
+of the actual allocation. This means that the remaining space at the end of the
+allocation is not cleared, potentially leaking code and/or data of a previous
+TA. The information gained by this attack is limited by the memory layout of
+the (compromised) TA performing the attack and the flags (i.e. is unloading the
+TA prevented when the last session is closed due to
+`TA_FLAG_INSTANCE_KEEP_ALIVE`) and layout of the attacked TA. Therefore we
+consider the risk to be
```

Comment:
...to be what? (same below)

Reply:
Let it up for the reader to decide, that makes it more like a storytale :smile: 

No seriously, copy/paste bug from the Riscure doc, I'll fix on Monday.

Reply:
It should end with "medium", but I think I', gonna remove that last sentence entirely. It doesn't really belong in the text at the security advisories page I think.